src/app/patient-dashboard/patient-info/update-patient-name.component.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, OnInit } from '@angular/core';
+import { AppFeatureAnalytics } from '../../shared/app-analytics/app-feature-analytics.service';
+import { PatientService } from '../patient.service';
+import { PersonResourceService } from '../../openmrs-api/person-resource.service';
+
+import { Patient } from '../../models/patient.model';
+
+@Component({
+  selector: 'update-patient-name',
+  templateUrl: './update-patient-name.component.html',
+  styleUrls: ['./update-patient-name.component.css']
+})
+
+export class UpdateNameComponent {
+    patient: Patient = new Patient({});
+    public display: boolean = false;
+    private patientName: any;
+    private patientuid: string;
+    private personNameError: string = null;
+    private preferOptions = [
+        { label: 'Yes', val: true },
+        { label: 'No', val: false }
+    ];
+
+    constructor(
+        private appFeatureAnalytics: AppFeatureAnalytics,
+        private patientService: PatientService,
+        private personResourceService: PersonResourceService
+    ) {}
+    public showDialog() {
+        this.display = true;
+    }
+
+    public updateName() {
+
+        let personNamePayload = {
+            names: [
+                {
```

Comment:
Ask Alfayo to help you out, because this will create multiple name obsjects in the database, whereas modifying the existing one would have been sufficient. We need the name uuid included.